tests/test_rigid_physics.py

```diff
@@ -430,3 +431,20 @@ def test_nonconvex_collision(show_viewer):
 
     if show_viewer:
         scene.viewer.stop()
+
+
+@pytest.mark.parametrize("xml_path", ["xml/test_equality.xml"])
+@pytest.mark.parametrize("gs_solver", [gs.constraint_solver.CG])
+@pytest.mark.parametrize("gs_integrator", [gs.integrator.Euler])
+@pytest.mark.parametrize("backend", [gs.cpu], indirect=True)
+def test_equality_joint(gs_sim, mj_sim):
+    # there is an equality constraint
+    np.testing.assert_allclose(gs_sim.rigid_solver.n_equalities, 1, 0.1)
+
+    qpos = np.array((0.0, -1.0))
+    qvel = np.array((1, -0.3))
+    simulate_and_check_mujoco_consistency(gs_sim, mj_sim, qpos, qvel, num_steps=150)
+
+    # check if the two joints are equal
+    gs_qpos = gs_sim.rigid_solver.qpos.to_numpy()[:, 0]
+    np.testing.assert_allclose(gs_qpos[0], gs_qpos[1], atol=1e-3)
```

Comment:
You cannot do better than that in terms of precision ? This is pretty bad.

Reply:
max difference is like 2.54518715e-05. Since this is not a hard constraint, we should not expect it to be very small.

Reply:
I see. But it is supposed to hard even if approximated as soft in practice. That is what the user would expect. Does increasing the simulation help to converge? I think it should be the case. If so, then increase the tolerance. I think 1e-5 would be good enough.

Reply:
Sure. Increase the simulation step and it can reach 1e-9.